From: =?utf-8?q?Guido_G=C3=BCnther?= <agx@sigxcpu.org>
Date: Thu, 16 Mar 2017 09:19:02 +0100
Subject: qemu: skip QMP probing of CPU definitions when missing

This unbreaks emulators that don't support this command such as
qemu-system-mips*.

Closes: #854125
---
 src/qemu/qemu_capabilities.c                       |   5 +
 src/qemu/qemu_capabilities.h                       |   1 +
 tests/qemucapabilitiesdata/caps_1.2.2.x86_64.xml   |   1 +
 tests/qemucapabilitiesdata/caps_1.3.1.x86_64.xml   |   1 +
 tests/qemucapabilitiesdata/caps_1.4.2.x86_64.xml   |   1 +
 tests/qemucapabilitiesdata/caps_1.5.3.x86_64.xml   |   1 +
 tests/qemucapabilitiesdata/caps_1.6.0.x86_64.xml   |   1 +
 tests/qemucapabilitiesdata/caps_1.7.0.x86_64.xml   |   1 +
 tests/qemucapabilitiesdata/caps_2.1.1.x86_64.xml   |   1 +
 tests/qemucapabilitiesdata/caps_2.4.0.x86_64.xml   |   1 +
 tests/qemucapabilitiesdata/caps_2.5.0.x86_64.xml   |   1 +
 .../caps_2.6.0-gicv2.aarch64.xml                   |   1 +
 .../caps_2.6.0-gicv3.aarch64.xml                   |   1 +
 tests/qemucapabilitiesdata/caps_2.6.0.ppc64le.xml  |   1 +
 tests/qemucapabilitiesdata/caps_2.6.0.x86_64.xml   |   1 +
 tests/qemucapabilitiesdata/caps_2.7.0.s390x.xml    |   1 +
 tests/qemucapabilitiesdata/caps_2.7.0.x86_64.xml   |   1 +
 tests/qemucapabilitiesdata/caps_2.8.0.s390x.xml    |   1 +
 tests/qemucapabilitiesdata/caps_2.8.0.x86_64.xml   |   1 +
 tests/qemucapabilitiesdata/caps_2.9.0.x86_64.xml   | 769 +++++++++++++++++++++
 20 files changed, 792 insertions(+)
 create mode 100644 tests/qemucapabilitiesdata/caps_2.9.0.x86_64.xml

diff --git a/src/qemu/qemu_capabilities.c b/src/qemu/qemu_capabilities.c
index 89e9dd4..69acd32 100644
--- a/src/qemu/qemu_capabilities.c
+++ b/src/qemu/qemu_capabilities.c
@@ -356,6 +356,7 @@ VIR_ENUM_IMPL(virQEMUCaps, QEMU_CAPS_LAST,
               "drive-iotune-group",
 
               "query-cpu-model-expansion", /* 245 */
+              "query-cpu-definitions",
     );
 
 
@@ -1520,6 +1521,7 @@ struct virQEMUCapsStringFlags virQEMUCapsCommands[] = {
     { "query-hotpluggable-cpus", QEMU_CAPS_QUERY_HOTPLUGGABLE_CPUS },
     { "query-qmp-schema", QEMU_CAPS_QUERY_QMP_SCHEMA },
     { "query-cpu-model-expansion", QEMU_CAPS_QUERY_CPU_MODEL_EXPANSION},
+    { "query-cpu-definitions", QEMU_CAPS_QUERY_CPU_DEFINITIONS},
 };
 
 struct virQEMUCapsStringFlags virQEMUCapsMigration[] = {
@@ -2786,6 +2788,9 @@ virQEMUCapsProbeQMPCPUDefinitions(virQEMUCapsPtr qemuCaps,
     int ret = -1;
     size_t i;
 
+    if (!virQEMUCapsGet(qemuCaps, QEMU_CAPS_QUERY_CPU_DEFINITIONS))
+        return 0;
+
     if ((ncpus = qemuMonitorGetCPUDefinitions(mon, &cpus)) < 0)
         return -1;
 
diff --git a/src/qemu/qemu_capabilities.h b/src/qemu/qemu_capabilities.h
index b5ad95e..0f0ae9c 100644
--- a/src/qemu/qemu_capabilities.h
+++ b/src/qemu/qemu_capabilities.h
@@ -392,6 +392,7 @@ typedef enum {
 
     /* 245 */
     QEMU_CAPS_QUERY_CPU_MODEL_EXPANSION, /* qmp query-cpu-model-expansion */
+    QEMU_CAPS_QUERY_CPU_DEFINITIONS, /* qmp query-cpu-definitions */
 
     QEMU_CAPS_LAST /* this must always be the last item */
 } virQEMUCapsFlags;
diff --git a/tests/qemucapabilitiesdata/caps_1.2.2.x86_64.xml b/tests/qemucapabilitiesdata/caps_1.2.2.x86_64.xml
index 6ea39c6..5b9913a 100644
--- a/tests/qemucapabilitiesdata/caps_1.2.2.x86_64.xml
+++ b/tests/qemucapabilitiesdata/caps_1.2.2.x86_64.xml
@@ -110,6 +110,7 @@
   <flag name='qxl.vram64_size_mb'/>
   <flag name='device-tray-moved-event'/>
   <flag name='display'/>
+  <flag name='query-cpu-definitions'/>
   <version>1002002</version>
   <kvmVersion>0</kvmVersion>
   <package></package>
diff --git a/tests/qemucapabilitiesdata/caps_1.3.1.x86_64.xml b/tests/qemucapabilitiesdata/caps_1.3.1.x86_64.xml
index aeb5cef..0bd307d 100644
--- a/tests/qemucapabilitiesdata/caps_1.3.1.x86_64.xml
+++ b/tests/qemucapabilitiesdata/caps_1.3.1.x86_64.xml
@@ -128,6 +128,7 @@
   <flag name='device-tray-moved-event'/>
   <flag name='nec-usb-xhci-ports'/>
   <flag name='display'/>
+  <flag name='query-cpu-definitions'/>
   <version>1003001</version>
   <kvmVersion>0</kvmVersion>
   <package></package>
diff --git a/tests/qemucapabilitiesdata/caps_1.4.2.x86_64.xml b/tests/qemucapabilitiesdata/caps_1.4.2.x86_64.xml
index 5b41244..ad044e7 100644
--- a/tests/qemucapabilitiesdata/caps_1.4.2.x86_64.xml
+++ b/tests/qemucapabilitiesdata/caps_1.4.2.x86_64.xml
@@ -129,6 +129,7 @@
   <flag name='device-tray-moved-event'/>
   <flag name='nec-usb-xhci-ports'/>
   <flag name='display'/>
+  <flag name='query-cpu-definitions'/>
   <version>1004002</version>
   <kvmVersion>0</kvmVersion>
   <package></package>
diff --git a/tests/qemucapabilitiesdata/caps_1.5.3.x86_64.xml b/tests/qemucapabilitiesdata/caps_1.5.3.x86_64.xml
index d7c0b17..a919ee4 100644
--- a/tests/qemucapabilitiesdata/caps_1.5.3.x86_64.xml
+++ b/tests/qemucapabilitiesdata/caps_1.5.3.x86_64.xml
@@ -140,6 +140,7 @@
   <flag name='nec-usb-xhci-ports'/>
   <flag name='display'/>
   <flag name='vhost-scsi'/>
+  <flag name='query-cpu-definitions'/>
   <version>1005003</version>
   <kvmVersion>0</kvmVersion>
   <package></package>
diff --git a/tests/qemucapabilitiesdata/caps_1.6.0.x86_64.xml b/tests/qemucapabilitiesdata/caps_1.6.0.x86_64.xml
index 25b9f56..51560a3 100644
--- a/tests/qemucapabilitiesdata/caps_1.6.0.x86_64.xml
+++ b/tests/qemucapabilitiesdata/caps_1.6.0.x86_64.xml
@@ -145,6 +145,7 @@
   <flag name='nec-usb-xhci-ports'/>
   <flag name='display'/>
   <flag name='vhost-scsi'/>
+  <flag name='query-cpu-definitions'/>
   <version>1006000</version>
   <kvmVersion>0</kvmVersion>
   <package></package>
diff --git a/tests/qemucapabilitiesdata/caps_1.7.0.x86_64.xml b/tests/qemucapabilitiesdata/caps_1.7.0.x86_64.xml
index 1323a0c..f4ee8d8 100644
--- a/tests/qemucapabilitiesdata/caps_1.7.0.x86_64.xml
+++ b/tests/qemucapabilitiesdata/caps_1.7.0.x86_64.xml
@@ -147,6 +147,7 @@
   <flag name='nec-usb-xhci-ports'/>
   <flag name='display'/>
   <flag name='vhost-scsi'/>
+  <flag name='query-cpu-definitions'/>
   <version>1007000</version>
   <kvmVersion>0</kvmVersion>
   <package></package>
diff --git a/tests/qemucapabilitiesdata/caps_2.1.1.x86_64.xml b/tests/qemucapabilitiesdata/caps_2.1.1.x86_64.xml
index f47aa84..bdc7c4a 100644
--- a/tests/qemucapabilitiesdata/caps_2.1.1.x86_64.xml
+++ b/tests/qemucapabilitiesdata/caps_2.1.1.x86_64.xml
@@ -162,6 +162,7 @@
   <flag name='drive-detect-zeroes'/>
   <flag name='display'/>
   <flag name='vhost-scsi'/>
+  <flag name='query-cpu-definitions'/>
   <version>2001001</version>
   <kvmVersion>0</kvmVersion>
   <package></package>
diff --git a/tests/qemucapabilitiesdata/caps_2.4.0.x86_64.xml b/tests/qemucapabilitiesdata/caps_2.4.0.x86_64.xml
index 9248a06..74ae9a3 100644
--- a/tests/qemucapabilitiesdata/caps_2.4.0.x86_64.xml
+++ b/tests/qemucapabilitiesdata/caps_2.4.0.x86_64.xml
@@ -183,6 +183,7 @@
   <flag name='virtio-vga'/>
   <flag name='vhost-scsi'/>
   <flag name='drive-iotune-group'/>
+  <flag name='query-cpu-definitions'/>
   <version>2004000</version>
   <kvmVersion>0</kvmVersion>
   <package></package>
diff --git a/tests/qemucapabilitiesdata/caps_2.5.0.x86_64.xml b/tests/qemucapabilitiesdata/caps_2.5.0.x86_64.xml
index 96e62d3..c7aee7a 100644
--- a/tests/qemucapabilitiesdata/caps_2.5.0.x86_64.xml
+++ b/tests/qemucapabilitiesdata/caps_2.5.0.x86_64.xml
@@ -189,6 +189,7 @@
   <flag name='query-qmp-schema'/>
   <flag name='vhost-scsi'/>
   <flag name='drive-iotune-group'/>
+  <flag name='query-cpu-definitions'/>
   <version>2005000</version>
   <kvmVersion>0</kvmVersion>
   <package></package>
diff --git a/tests/qemucapabilitiesdata/caps_2.6.0-gicv2.aarch64.xml b/tests/qemucapabilitiesdata/caps_2.6.0-gicv2.aarch64.xml
index 0aed651..f45560b 100644
--- a/tests/qemucapabilitiesdata/caps_2.6.0-gicv2.aarch64.xml
+++ b/tests/qemucapabilitiesdata/caps_2.6.0-gicv2.aarch64.xml
@@ -164,6 +164,7 @@
   <flag name='query-qmp-schema'/>
   <flag name='vhost-scsi'/>
   <flag name='drive-iotune-group'/>
+  <flag name='query-cpu-definitions'/>
   <version>2006000</version>
   <kvmVersion>0</kvmVersion>
   <package></package>
diff --git a/tests/qemucapabilitiesdata/caps_2.6.0-gicv3.aarch64.xml b/tests/qemucapabilitiesdata/caps_2.6.0-gicv3.aarch64.xml
index 1041a12..721c97b 100644
--- a/tests/qemucapabilitiesdata/caps_2.6.0-gicv3.aarch64.xml
+++ b/tests/qemucapabilitiesdata/caps_2.6.0-gicv3.aarch64.xml
@@ -164,6 +164,7 @@
   <flag name='query-qmp-schema'/>
   <flag name='vhost-scsi'/>
   <flag name='drive-iotune-group'/>
+  <flag name='query-cpu-definitions'/>
   <version>2006000</version>
   <kvmVersion>0</kvmVersion>
   <package></package>
diff --git a/tests/qemucapabilitiesdata/caps_2.6.0.ppc64le.xml b/tests/qemucapabilitiesdata/caps_2.6.0.ppc64le.xml
index 92e2781..f1c5105 100644
--- a/tests/qemucapabilitiesdata/caps_2.6.0.ppc64le.xml
+++ b/tests/qemucapabilitiesdata/caps_2.6.0.ppc64le.xml
@@ -159,6 +159,7 @@
   <flag name='query-qmp-schema'/>
   <flag name='vhost-scsi'/>
   <flag name='drive-iotune-group'/>
+  <flag name='query-cpu-definitions'/>
   <version>2006000</version>
   <kvmVersion>0</kvmVersion>
   <package></package>
diff --git a/tests/qemucapabilitiesdata/caps_2.6.0.x86_64.xml b/tests/qemucapabilitiesdata/caps_2.6.0.x86_64.xml
index faddd50..46aae4f 100644
--- a/tests/qemucapabilitiesdata/caps_2.6.0.x86_64.xml
+++ b/tests/qemucapabilitiesdata/caps_2.6.0.x86_64.xml
@@ -198,6 +198,7 @@
   <flag name='query-qmp-schema'/>
   <flag name='vhost-scsi'/>
   <flag name='drive-iotune-group'/>
+  <flag name='query-cpu-definitions'/>
   <version>2006000</version>
   <kvmVersion>0</kvmVersion>
   <package></package>
diff --git a/tests/qemucapabilitiesdata/caps_2.7.0.s390x.xml b/tests/qemucapabilitiesdata/caps_2.7.0.s390x.xml
index af21017..4c45b38 100644
--- a/tests/qemucapabilitiesdata/caps_2.7.0.s390x.xml
+++ b/tests/qemucapabilitiesdata/caps_2.7.0.s390x.xml
@@ -127,6 +127,7 @@
   <flag name='gluster.debug_level'/>
   <flag name='vhost-scsi'/>
   <flag name='drive-iotune-group'/>
+  <flag name='query-cpu-definitions'/>
   <version>2007000</version>
   <kvmVersion>0</kvmVersion>
   <package></package>
diff --git a/tests/qemucapabilitiesdata/caps_2.7.0.x86_64.xml b/tests/qemucapabilitiesdata/caps_2.7.0.x86_64.xml
index c789f0e..e57a550 100644
--- a/tests/qemucapabilitiesdata/caps_2.7.0.x86_64.xml
+++ b/tests/qemucapabilitiesdata/caps_2.7.0.x86_64.xml
@@ -200,6 +200,7 @@
   <flag name='gluster.debug_level'/>
   <flag name='vhost-scsi'/>
   <flag name='drive-iotune-group'/>
+  <flag name='query-cpu-definitions'/>
   <version>2007000</version>
   <kvmVersion>0</kvmVersion>
   <package> (v2.7.0)</package>
diff --git a/tests/qemucapabilitiesdata/caps_2.8.0.s390x.xml b/tests/qemucapabilitiesdata/caps_2.8.0.s390x.xml
index c4c9bf9..3f41be2 100644
--- a/tests/qemucapabilitiesdata/caps_2.8.0.s390x.xml
+++ b/tests/qemucapabilitiesdata/caps_2.8.0.s390x.xml
@@ -129,6 +129,7 @@
   <flag name='vhost-scsi'/>
   <flag name='drive-iotune-group'/>
   <flag name='query-cpu-model-expansion'/>
+  <flag name='query-cpu-definitions'/>
   <version>2007093</version>
   <kvmVersion>0</kvmVersion>
   <package></package>
diff --git a/tests/qemucapabilitiesdata/caps_2.8.0.x86_64.xml b/tests/qemucapabilitiesdata/caps_2.8.0.x86_64.xml
index 9757bd2..beee7ed 100644
--- a/tests/qemucapabilitiesdata/caps_2.8.0.x86_64.xml
+++ b/tests/qemucapabilitiesdata/caps_2.8.0.x86_64.xml
@@ -198,6 +198,7 @@
   <flag name='gluster.debug_level'/>
   <flag name='vhost-scsi'/>
   <flag name='drive-iotune-group'/>
+  <flag name='query-cpu-definitions'/>
   <version>2007093</version>
   <kvmVersion>0</kvmVersion>
   <package> (v2.8.0-rc3-dirty)</package>
diff --git a/tests/qemucapabilitiesdata/caps_2.9.0.x86_64.xml b/tests/qemucapabilitiesdata/caps_2.9.0.x86_64.xml
new file mode 100644
index 0000000..d975322
--- /dev/null
+++ b/tests/qemucapabilitiesdata/caps_2.9.0.x86_64.xml
@@ -0,0 +1,769 @@
+<qemuCaps>
+  <qemuctime>0</qemuctime>
+  <selfctime>0</selfctime>
+  <selfvers>0</selfvers>
+  <usedQMP/>
+  <flag name='kvm'/>
+  <flag name='mem-path'/>
+  <flag name='drive-serial'/>
+  <flag name='chardev'/>
+  <flag name='monitor-json'/>
+  <flag name='sdl'/>
+  <flag name='netdev'/>
+  <flag name='rtc'/>
+  <flag name='vhost-net'/>
+  <flag name='no-hpet'/>
+  <flag name='no-kvm-pit'/>
+  <flag name='pci-configfd'/>
+  <flag name='nodefconfig'/>
+  <flag name='boot-menu'/>
+  <flag name='fsdev'/>
+  <flag name='name-process'/>
+  <flag name='smbios-type'/>
+  <flag name='spice'/>
+  <flag name='vga-none'/>
+  <flag name='boot-index'/>
+  <flag name='hda-duplex'/>
+  <flag name='drive-aio'/>
+  <flag name='pci-bootindex'/>
+  <flag name='ccid-emulated'/>
+  <flag name='ccid-passthru'/>
+  <flag name='chardev-spicevmc'/>
+  <flag name='virtio-tx-alg'/>
+  <flag name='pci-multifunction'/>
+  <flag name='virtio-blk-pci.ioeventfd'/>
+  <flag name='sga'/>
+  <flag name='virtio-blk-pci.event_idx'/>
+  <flag name='virtio-net-pci.event_idx'/>
+  <flag name='cache-directsync'/>
+  <flag name='piix3-usb-uhci'/>
+  <flag name='piix4-usb-uhci'/>
+  <flag name='usb-ehci'/>
+  <flag name='ich9-usb-ehci1'/>
+  <flag name='vt82c686b-usb-uhci'/>
+  <flag name='pci-ohci'/>
+  <flag name='usb-redir'/>
+  <flag name='usb-hub'/>
+  <flag name='no-shutdown'/>
+  <flag name='cache-unsafe'/>
+  <flag name='ich9-ahci'/>
+  <flag name='no-acpi'/>
+  <flag name='fsdev-readonly'/>
+  <flag name='virtio-blk-pci.scsi'/>
+  <flag name='drive-copy-on-read'/>
+  <flag name='fsdev-writeout'/>
+  <flag name='drive-iotune'/>
+  <flag name='system_wakeup'/>
+  <flag name='scsi-disk.channel'/>
+  <flag name='scsi-block'/>
+  <flag name='transaction'/>
+  <flag name='block-job-async'/>
+  <flag name='scsi-cd'/>
+  <flag name='ide-cd'/>
+  <flag name='no-user-config'/>
+  <flag name='hda-micro'/>
+  <flag name='dump-guest-memory'/>
+  <flag name='nec-usb-xhci'/>
+  <flag name='balloon-event'/>
+  <flag name='bridge'/>
+  <flag name='lsi'/>
+  <flag name='virtio-scsi-pci'/>
+  <flag name='blockio'/>
+  <flag name='disable-s3'/>
+  <flag name='disable-s4'/>
+  <flag name='usb-redir.filter'/>
+  <flag name='ide-drive.wwn'/>
+  <flag name='scsi-disk.wwn'/>
+  <flag name='seccomp-sandbox'/>
+  <flag name='reboot-timeout'/>
+  <flag name='dump-guest-core'/>
+  <flag name='seamless-migration'/>
+  <flag name='block-commit'/>
+  <flag name='vnc'/>
+  <flag name='drive-mirror'/>
+  <flag name='usb-redir.bootindex'/>
+  <flag name='usb-host.bootindex'/>
+  <flag name='blockdev-snapshot-sync'/>
+  <flag name='qxl'/>
+  <flag name='VGA'/>
+  <flag name='cirrus-vga'/>
+  <flag name='vmware-svga'/>
+  <flag name='device-video-primary'/>
+  <flag name='usb-serial'/>
+  <flag name='usb-net'/>
+  <flag name='add-fd'/>
+  <flag name='nbd-server'/>
+  <flag name='virtio-rng'/>
+  <flag name='rng-random'/>
+  <flag name='rng-egd'/>
+  <flag name='dtb'/>
+  <flag name='megasas'/>
+  <flag name='ipv6-migration'/>
+  <flag name='machine-opt'/>
+  <flag name='machine-usb-opt'/>
+  <flag name='tpm-passthrough'/>
+  <flag name='tpm-tis'/>
+  <flag name='pci-bridge'/>
+  <flag name='vfio-pci'/>
+  <flag name='vfio-pci.bootindex'/>
+  <flag name='scsi-generic'/>
+  <flag name='scsi-generic.bootindex'/>
+  <flag name='mem-merge'/>
+  <flag name='vnc-websocket'/>
+  <flag name='drive-discard'/>
+  <flag name='mlock'/>
+  <flag name='vnc-share-policy'/>
+  <flag name='device-del-event'/>
+  <flag name='dmi-to-pci-bridge'/>
+  <flag name='i440fx-pci-hole64-size'/>
+  <flag name='q35-pci-hole64-size'/>
+  <flag name='usb-storage'/>
+  <flag name='usb-storage.removable'/>
+  <flag name='virtio-mmio'/>
+  <flag name='ich9-intel-hda'/>
+  <flag name='kvm-pit-lost-tick-policy'/>
+  <flag name='boot-strict'/>
+  <flag name='pvpanic'/>
+  <flag name='spice-file-xfer-disable'/>
+  <flag name='spiceport'/>
+  <flag name='usb-kbd'/>
+  <flag name='host-pci-multidomain'/>
+  <flag name='msg-timestamp'/>
+  <flag name='active-commit'/>
+  <flag name='change-backing-file'/>
+  <flag name='memory-backend-ram'/>
+  <flag name='numa'/>
+  <flag name='memory-backend-file'/>
+  <flag name='usb-audio'/>
+  <flag name='rtc-reset-reinjection'/>
+  <flag name='splash-timeout'/>
+  <flag name='iothread'/>
+  <flag name='migrate-rdma'/>
+  <flag name='ivshmem'/>
+  <flag name='drive-iotune-max'/>
+  <flag name='VGA.vgamem_mb'/>
+  <flag name='vmware-svga.vgamem_mb'/>
+  <flag name='qxl.vgamem_mb'/>
+  <flag name='pc-dimm'/>
+  <flag name='machine-vmport-opt'/>
+  <flag name='aes-key-wrap'/>
+  <flag name='dea-key-wrap'/>
+  <flag name='pci-serial'/>
+  <flag name='vhost-user-multiqueue'/>
+  <flag name='migration-event'/>
+  <flag name='ioh3420'/>
+  <flag name='x3130-upstream'/>
+  <flag name='xio3130-downstream'/>
+  <flag name='rtl8139'/>
+  <flag name='e1000'/>
+  <flag name='virtio-net'/>
+  <flag name='gic-version'/>
+  <flag name='incoming-defer'/>
+  <flag name='virtio-gpu'/>
+  <flag name='virtio-gpu.virgl'/>
+  <flag name='virtio-keyboard'/>
+  <flag name='virtio-mouse'/>
+  <flag name='virtio-tablet'/>
+  <flag name='virtio-input-host'/>
+  <flag name='chardev-file-append'/>
+  <flag name='ich9-disable-s3'/>
+  <flag name='ich9-disable-s4'/>
+  <flag name='vserport-change-event'/>
+  <flag name='virtio-balloon-pci.deflate-on-oom'/>
+  <flag name='mptsas1068'/>
+  <flag name='spice-gl'/>
+  <flag name='qxl.vram64_size_mb'/>
+  <flag name='chardev-logfile'/>
+  <flag name='debug-threads'/>
+  <flag name='secret'/>
+  <flag name='pxb'/>
+  <flag name='pxb-pcie'/>
+  <flag name='device-tray-moved-event'/>
+  <flag name='nec-usb-xhci-ports'/>
+  <flag name='virtio-scsi-pci.iothread'/>
+  <flag name='name-guest'/>
+  <flag name='qxl.max_outputs'/>
+  <flag name='spice-unix'/>
+  <flag name='drive-detect-zeroes'/>
+  <flag name='tls-creds-x509'/>
+  <flag name='display'/>
+  <flag name='intel-iommu'/>
+  <flag name='smm'/>
+  <flag name='virtio-pci-disable-legacy'/>
+  <flag name='query-hotpluggable-cpus'/>
+  <flag name='virtio-net.rx_queue_size'/>
+  <flag name='virtio-vga'/>
+  <flag name='drive-iotune-max-length'/>
+  <flag name='ivshmem-plain'/>
+  <flag name='ivshmem-doorbell'/>
+  <flag name='query-qmp-schema'/>
+  <flag name='gluster.debug_level'/>
+  <flag name='vhost-scsi'/>
+  <flag name='drive-iotune-group'/>
+  <flag name='query-cpu-model-expansion'/>
+  <flag name='virtio-net.host_mtu'/>
+  <flag name='query-cpu-definitions'/>
+  <version>2008050</version>
+  <kvmVersion>0</kvmVersion>
+  <package> (v2.8.0-1961-g5b10b94bd5)</package>
+  <arch>x86_64</arch>
+  <hostCPU type='kvm' model='base'>
+    <property name='phys-bits' type='number' value='0'/>
+    <property name='core-id' type='number' value='-1'/>
+    <property name='xlevel' type='number' value='2147483656'/>
+    <property name='cmov' type='boolean' value='true'/>
+    <property name='ia64' type='boolean' value='false'/>
+    <property name='aes' type='boolean' value='true'/>
+    <property name='mmx' type='boolean' value='true'/>
+    <property name='arat' type='boolean' value='true'/>
+    <property name='rdpid' type='boolean' value='false'/>
+    <property name='pause-filter' type='boolean' value='false'/>
+    <property name='xsavec' type='boolean' value='true'/>
+    <property name='osxsave' type='boolean' value='false'/>
+    <property name='tsc-frequency' type='number' value='0'/>
+    <property name='xd' type='boolean' value='true'/>
+    <property name='hv-vendor-id' type='string' value=''/>
+    <property name='kvm-asyncpf' type='boolean' value='true'/>
+    <property name='kvm_asyncpf' type='boolean' value='true'/>
+    <property name='perfctr_core' type='boolean' value='false'/>
+    <property name='perfctr-core' type='boolean' value='false'/>
+    <property name='mpx' type='boolean' value='true'/>
+    <property name='avx512cd' type='boolean' value='false'/>
+    <property name='decodeassists' type='boolean' value='false'/>
+    <property name='pbe' type='boolean' value='false'/>
+    <property name='sse4_1' type='boolean' value='true'/>
+    <property name='sse4.1' type='boolean' value='true'/>
+    <property name='sse4-1' type='boolean' value='true'/>
+    <property name='family' type='number' value='6'/>
+    <property name='vmware-cpuid-freq' type='boolean' value='true'/>
+    <property name='avx512f' type='boolean' value='false'/>
+    <property name='xcrypt' type='boolean' value='false'/>
+    <property name='hv-runtime' type='boolean' value='false'/>
+    <property name='msr' type='boolean' value='true'/>
+    <property name='mce' type='boolean' value='true'/>
+    <property name='mca' type='boolean' value='true'/>
+    <property name='thread-id' type='number' value='-1'/>
+    <property name='min-level' type='number' value='13'/>
+    <property name='xgetbv1' type='boolean' value='true'/>
+    <property name='cid' type='boolean' value='false'/>
+    <property name='hv-relaxed' type='boolean' value='false'/>
+    <property name='fxsr' type='boolean' value='true'/>
+    <property name='ds' type='boolean' value='false'/>
+    <property name='hv-crash' type='boolean' value='false'/>
+    <property name='xsaveopt' type='boolean' value='true'/>
+    <property name='xtpr' type='boolean' value='false'/>
+    <property name='avx512-vpopcntdq' type='boolean' value='false'/>
+    <property name='phe' type='boolean' value='false'/>
+    <property name='avx512vl' type='boolean' value='false'/>
+    <property name='extapic' type='boolean' value='false'/>
+    <property name='3dnowprefetch' type='boolean' value='true'/>
+    <property name='cr8legacy' type='boolean' value='false'/>
+    <property name='cpuid-0xb' type='boolean' value='true'/>
+    <property name='xcrypt-en' type='boolean' value='false'/>
+    <property name='kvm_pv_eoi' type='boolean' value='true'/>
+    <property name='apic-id' type='number' value='4294967295'/>
+    <property name='pn' type='boolean' value='false'/>
+    <property name='dca' type='boolean' value='false'/>
+    <property name='vendor' type='string' value='GenuineIntel'/>
+    <property name='pku' type='boolean' value='false'/>
+    <property name='smx' type='boolean' value='false'/>
+    <property name='cmp-legacy' type='boolean' value='false'/>
+    <property name='cmp_legacy' type='boolean' value='false'/>
+    <property name='avx512-4fmaps' type='boolean' value='false'/>
+    <property name='vmcb-clean' type='boolean' value='false'/>
+    <property name='vmcb_clean' type='boolean' value='false'/>
+    <property name='3dnowext' type='boolean' value='false'/>
+    <property name='hle' type='boolean' value='true'/>
+    <property name='npt' type='boolean' value='false'/>
+    <property name='memory' type='string' value='/machine/unattached/system[0]'/>
+    <property name='clwb' type='boolean' value='false'/>
+    <property name='lbrv' type='boolean' value='false'/>
+    <property name='adx' type='boolean' value='true'/>
+    <property name='ss' type='boolean' value='true'/>
+    <property name='pni' type='boolean' value='true'/>
+    <property name='svm_lock' type='boolean' value='false'/>
+    <property name='svm-lock' type='boolean' value='false'/>
+    <property name='smep' type='boolean' value='true'/>
+    <property name='pfthreshold' type='boolean' value='false'/>
+    <property name='smap' type='boolean' value='true'/>
+    <property name='x2apic' type='boolean' value='true'/>
+    <property name='avx512vbmi' type='boolean' value='false'/>
+    <property name='hv-stimer' type='boolean' value='false'/>
+    <property name='i64' type='boolean' value='true'/>
+    <property name='flushbyasid' type='boolean' value='false'/>
+    <property name='f16c' type='boolean' value='true'/>
+    <property name='ace2-en' type='boolean' value='false'/>
+    <property name='pat' type='boolean' value='true'/>
+    <property name='pae' type='boolean' value='true'/>
+    <property name='sse' type='boolean' value='true'/>
+    <property name='phe-en' type='boolean' value='false'/>
+    <property name='kvm-nopiodelay' type='boolean' value='true'/>
+    <property name='kvm_nopiodelay' type='boolean' value='true'/>
+    <property name='tm' type='boolean' value='false'/>
+    <property name='kvmclock-stable-bit' type='boolean' value='true'/>
+    <property name='hypervisor' type='boolean' value='true'/>
+    <property name='socket-id' type='number' value='-1'/>
+    <property name='pcommit' type='boolean' value='false'/>
+    <property name='syscall' type='boolean' value='true'/>
+    <property name='level' type='number' value='13'/>
+    <property name='avx512dq' type='boolean' value='false'/>
+    <property name='svm' type='boolean' value='false'/>
+    <property name='full-cpuid-auto-level' type='boolean' value='true'/>
+    <property name='hv-reset' type='boolean' value='false'/>
+    <property name='invtsc' type='boolean' value='false'/>
+    <property name='sse3' type='boolean' value='true'/>
+    <property name='sse2' type='boolean' value='true'/>
+    <property name='est' type='boolean' value='false'/>
+    <property name='avx512ifma' type='boolean' value='false'/>
+    <property name='tm2' type='boolean' value='false'/>
+    <property name='kvm-pv-eoi' type='boolean' value='true'/>
+    <property name='cx8' type='boolean' value='true'/>
+    <property name='kvm-mmu' type='boolean' value='false'/>
+    <property name='kvm_mmu' type='boolean' value='false'/>
+    <property name='sse4_2' type='boolean' value='true'/>
+    <property name='sse4.2' type='boolean' value='true'/>
+    <property name='sse4-2' type='boolean' value='true'/>
+    <property name='pge' type='boolean' value='true'/>
+    <property name='fill-mtrr-mask' type='boolean' value='true'/>
+    <property name='pdcm' type='boolean' value='false'/>
+    <property name='nodeid_msr' type='boolean' value='false'/>
+    <property name='model' type='number' value='94'/>
+    <property name='movbe' type='boolean' value='true'/>
+    <property name='nrip-save' type='boolean' value='false'/>
+    <property name='nrip_save' type='boolean' value='false'/>
+    <property name='sse4a' type='boolean' value='false'/>
+    <property name='ssse3' type='boolean' value='true'/>
+    <property name='kvm_pv_unhalt' type='boolean' value='true'/>
+    <property name='invpcid' type='boolean' value='true'/>
+    <property name='pdpe1gb' type='boolean' value='true'/>
+    <property name='tsc-deadline' type='boolean' value='true'/>
+    <property name='fma' type='boolean' value='true'/>
+    <property name='cx16' type='boolean' value='true'/>
+    <property name='de' type='boolean' value='true'/>
+    <property name='enforce' type='boolean' value='false'/>
+    <property name='stepping' type='number' value='3'/>
+    <property name='xsave' type='boolean' value='true'/>
+    <property name='clflush' type='boolean' value='true'/>
+    <property name='skinit' type='boolean' value='false'/>
+    <property name='tce' type='boolean' value='false'/>
+    <property name='tsc' type='boolean' value='true'/>
+    <property name='fpu' type='boolean' value='true'/>
+    <property name='ds-cpl' type='boolean' value='false'/>
+    <property name='ds_cpl' type='boolean' value='false'/>
+    <property name='ibs' type='boolean' value='false'/>
+    <property name='host-phys-bits' type='boolean' value='false'/>
+    <property name='fma4' type='boolean' value='false'/>
+    <property name='la57' type='boolean' value='false'/>
+    <property name='osvw' type='boolean' value='false'/>
+    <property name='check' type='boolean' value='true'/>
+    <property name='hv-spinlocks' type='number' value='-1'/>
+    <property name='pmm' type='boolean' value='false'/>
+    <property name='apic' type='boolean' value='true'/>
+    <property name='pmu' type='boolean' value='false'/>
+    <property name='min-xlevel2' type='number' value='0'/>
+    <property name='tsc-adjust' type='boolean' value='true'/>
+    <property name='tsc_adjust' type='boolean' value='true'/>
+    <property name='kvm-steal-time' type='boolean' value='true'/>
+    <property name='kvm_steal_time' type='boolean' value='true'/>
+    <property name='kvmclock' type='boolean' value='true'/>
+    <property name='l3-cache' type='boolean' value='true'/>
+    <property name='lwp' type='boolean' value='false'/>
+    <property name='xop' type='boolean' value='false'/>
+    <property name='avx' type='boolean' value='true'/>
+    <property name='ospke' type='boolean' value='false'/>
+    <property name='ace2' type='boolean' value='false'/>
+    <property name='acpi' type='boolean' value='false'/>
+    <property name='avx512bw' type='boolean' value='false'/>
+    <property name='hv-vapic' type='boolean' value='false'/>
+    <property name='fsgsbase' type='boolean' value='true'/>
+    <property name='ht' type='boolean' value='false'/>
+    <property name='nx' type='boolean' value='true'/>
+    <property name='pclmulqdq' type='boolean' value='true'/>
+    <property name='mmxext' type='boolean' value='false'/>
+    <property name='popcnt' type='boolean' value='true'/>
+    <property name='xsaves' type='boolean' value='true'/>
+    <property name='lm' type='boolean' value='true'/>
+    <property name='umip' type='boolean' value='false'/>
+    <property name='avx2' type='boolean' value='true'/>
+    <property name='pse' type='boolean' value='true'/>
+    <property name='sep' type='boolean' value='true'/>
+    <property name='pclmuldq' type='boolean' value='true'/>
+    <property name='nodeid-msr' type='boolean' value='false'/>
+    <property name='kvm' type='boolean' value='true'/>
+    <property name='misalignsse' type='boolean' value='false'/>
+    <property name='min-xlevel' type='number' value='2147483656'/>
+    <property name='bmi2' type='boolean' value='true'/>
+    <property name='bmi1' type='boolean' value='true'/>
+    <property name='kvm-pv-unhalt' type='boolean' value='true'/>
+    <property name='realized' type='boolean' value='false'/>
+    <property name='tsc_scale' type='boolean' value='false'/>
+    <property name='tsc-scale' type='boolean' value='false'/>
+    <property name='topoext' type='boolean' value='false'/>
+    <property name='hv-vpindex' type='boolean' value='false'/>
+    <property name='xlevel2' type='number' value='0'/>
+    <property name='clflushopt' type='boolean' value='true'/>
+    <property name='monitor' type='boolean' value='false'/>
+    <property name='avx512er' type='boolean' value='false'/>
+    <property name='pmm-en' type='boolean' value='false'/>
+    <property name='pcid' type='boolean' value='true'/>
+    <property name='3dnow' type='boolean' value='false'/>
+    <property name='erms' type='boolean' value='true'/>
+    <property name='lahf-lm' type='boolean' value='true'/>
+    <property name='lahf_lm' type='boolean' value='true'/>
+    <property name='xstore' type='boolean' value='false'/>
+    <property name='hv-synic' type='boolean' value='false'/>
+    <property name='fxsr-opt' type='boolean' value='false'/>
+    <property name='fxsr_opt' type='boolean' value='false'/>
+    <property name='rtm' type='boolean' value='true'/>
+    <property name='lmce' type='boolean' value='true'/>
+    <property name='hv-time' type='boolean' value='false'/>
+    <property name='perfctr-nb' type='boolean' value='false'/>
+    <property name='perfctr_nb' type='boolean' value='false'/>
+    <property name='ffxsr' type='boolean' value='false'/>
+    <property name='rdrand' type='boolean' value='true'/>
+    <property name='rdseed' type='boolean' value='true'/>
+    <property name='avx512-4vnniw' type='boolean' value='false'/>
+    <property name='vme' type='boolean' value='true'/>
+    <property name='vmx' type='boolean' value='true'/>
+    <property name='dtes64' type='boolean' value='false'/>
+    <property name='mtrr' type='boolean' value='true'/>
+    <property name='rdtscp' type='boolean' value='true'/>
+    <property name='pse36' type='boolean' value='true'/>
+    <property name='tbm' type='boolean' value='false'/>
+    <property name='wdt' type='boolean' value='false'/>
+    <property name='pause_filter' type='boolean' value='false'/>
+    <property name='model-id' type='string' value='Intel(R) Xeon(R) CPU E3-1245 v5 @ 3.50GHz'/>
+    <property name='sha-ni' type='boolean' value='false'/>
+    <property name='abm' type='boolean' value='true'/>
+    <property name='avx512pf' type='boolean' value='false'/>
+    <property name='xstore-en' type='boolean' value='false'/>
+  </hostCPU>
+  <hostCPU type='tcg' model='base'>
+    <property name='phys-bits' type='number' value='0'/>
+    <property name='core-id' type='number' value='-1'/>
+    <property name='xlevel' type='number' value='2147483658'/>
+    <property name='cmov' type='boolean' value='true'/>
+    <property name='ia64' type='boolean' value='false'/>
+    <property name='aes' type='boolean' value='true'/>
+    <property name='mmx' type='boolean' value='true'/>
+    <property name='arat' type='boolean' value='true'/>
+    <property name='rdpid' type='boolean' value='false'/>
+    <property name='pause-filter' type='boolean' value='false'/>
+    <property name='xsavec' type='boolean' value='false'/>
+    <property name='osxsave' type='boolean' value='false'/>
+    <property name='tsc-frequency' type='number' value='0'/>
+    <property name='xd' type='boolean' value='true'/>
+    <property name='hv-vendor-id' type='string' value=''/>
+    <property name='kvm-asyncpf' type='boolean' value='false'/>
+    <property name='kvm_asyncpf' type='boolean' value='false'/>
+    <property name='perfctr_core' type='boolean' value='false'/>
+    <property name='perfctr-core' type='boolean' value='false'/>
+    <property name='mpx' type='boolean' value='true'/>
+    <property name='avx512cd' type='boolean' value='false'/>
+    <property name='decodeassists' type='boolean' value='false'/>
+    <property name='pbe' type='boolean' value='false'/>
+    <property name='sse4_1' type='boolean' value='true'/>
+    <property name='sse4.1' type='boolean' value='true'/>
+    <property name='sse4-1' type='boolean' value='true'/>
+    <property name='family' type='number' value='6'/>
+    <property name='vmware-cpuid-freq' type='boolean' value='true'/>
+    <property name='avx512f' type='boolean' value='false'/>
+    <property name='xcrypt' type='boolean' value='false'/>
+    <property name='hv-runtime' type='boolean' value='false'/>
+    <property name='msr' type='boolean' value='true'/>
+    <property name='mce' type='boolean' value='true'/>
+    <property name='mca' type='boolean' value='true'/>
+    <property name='thread-id' type='number' value='-1'/>
+    <property name='min-level' type='number' value='13'/>
+    <property name='xgetbv1' type='boolean' value='true'/>
+    <property name='cid' type='boolean' value='false'/>
+    <property name='hv-relaxed' type='boolean' value='false'/>
+    <property name='fxsr' type='boolean' value='true'/>
+    <property name='ds' type='boolean' value='false'/>
+    <property name='hv-crash' type='boolean' value='false'/>
+    <property name='xsaveopt' type='boolean' value='true'/>
+    <property name='xtpr' type='boolean' value='false'/>
+    <property name='avx512-vpopcntdq' type='boolean' value='false'/>
+    <property name='phe' type='boolean' value='false'/>
+    <property name='avx512vl' type='boolean' value='false'/>
+    <property name='extapic' type='boolean' value='false'/>
+    <property name='3dnowprefetch' type='boolean' value='false'/>
+    <property name='cr8legacy' type='boolean' value='true'/>
+    <property name='cpuid-0xb' type='boolean' value='true'/>
+    <property name='xcrypt-en' type='boolean' value='false'/>
+    <property name='kvm_pv_eoi' type='boolean' value='false'/>
+    <property name='apic-id' type='number' value='4294967295'/>
+    <property name='pn' type='boolean' value='false'/>
+    <property name='dca' type='boolean' value='false'/>
+    <property name='vendor' type='string' value='AuthenticAMD'/>
+    <property name='pku' type='boolean' value='true'/>
+    <property name='smx' type='boolean' value='false'/>
+    <property name='cmp-legacy' type='boolean' value='false'/>
+    <property name='cmp_legacy' type='boolean' value='false'/>
+    <property name='avx512-4fmaps' type='boolean' value='false'/>
+    <property name='vmcb-clean' type='boolean' value='false'/>
+    <property name='vmcb_clean' type='boolean' value='false'/>
+    <property name='3dnowext' type='boolean' value='true'/>
+    <property name='hle' type='boolean' value='false'/>
+    <property name='npt' type='boolean' value='false'/>
+    <property name='memory' type='string' value='/machine/unattached/system[0]'/>
+    <property name='clwb' type='boolean' value='true'/>
+    <property name='lbrv' type='boolean' value='false'/>
+    <property name='adx' type='boolean' value='true'/>
+    <property name='ss' type='boolean' value='true'/>
+    <property name='pni' type='boolean' value='true'/>
+    <property name='svm_lock' type='boolean' value='false'/>
+    <property name='svm-lock' type='boolean' value='false'/>
+    <property name='smep' type='boolean' value='true'/>
+    <property name='pfthreshold' type='boolean' value='false'/>
+    <property name='smap' type='boolean' value='true'/>
+    <property name='x2apic' type='boolean' value='false'/>
+    <property name='avx512vbmi' type='boolean' value='false'/>
+    <property name='hv-stimer' type='boolean' value='false'/>
+    <property name='i64' type='boolean' value='true'/>
+    <property name='flushbyasid' type='boolean' value='false'/>
+    <property name='f16c' type='boolean' value='false'/>
+    <property name='ace2-en' type='boolean' value='false'/>
+    <property name='pat' type='boolean' value='true'/>
+    <property name='pae' type='boolean' value='true'/>
+    <property name='sse' type='boolean' value='true'/>
+    <property name='phe-en' type='boolean' value='false'/>
+    <property name='kvm-nopiodelay' type='boolean' value='false'/>
+    <property name='kvm_nopiodelay' type='boolean' value='false'/>
+    <property name='tm' type='boolean' value='false'/>
+    <property name='kvmclock-stable-bit' type='boolean' value='false'/>
+    <property name='hypervisor' type='boolean' value='true'/>
+    <property name='socket-id' type='number' value='-1'/>
+    <property name='pcommit' type='boolean' value='true'/>
+    <property name='syscall' type='boolean' value='true'/>
+    <property name='level' type='number' value='13'/>
+    <property name='avx512dq' type='boolean' value='false'/>
+    <property name='svm' type='boolean' value='true'/>
+    <property name='full-cpuid-auto-level' type='boolean' value='true'/>
+    <property name='hv-reset' type='boolean' value='false'/>
+    <property name='invtsc' type='boolean' value='false'/>
+    <property name='sse3' type='boolean' value='true'/>
+    <property name='sse2' type='boolean' value='true'/>
+    <property name='est' type='boolean' value='false'/>
+    <property name='avx512ifma' type='boolean' value='false'/>
+    <property name='tm2' type='boolean' value='false'/>
+    <property name='kvm-pv-eoi' type='boolean' value='false'/>
+    <property name='cx8' type='boolean' value='true'/>
+    <property name='kvm-mmu' type='boolean' value='false'/>
+    <property name='kvm_mmu' type='boolean' value='false'/>
+    <property name='sse4_2' type='boolean' value='true'/>
+    <property name='sse4.2' type='boolean' value='true'/>
+    <property name='sse4-2' type='boolean' value='true'/>
+    <property name='pge' type='boolean' value='true'/>
+    <property name='fill-mtrr-mask' type='boolean' value='true'/>
+    <property name='pdcm' type='boolean' value='false'/>
+    <property name='nodeid_msr' type='boolean' value='false'/>
+    <property name='model' type='number' value='6'/>
+    <property name='movbe' type='boolean' value='true'/>
+    <property name='nrip-save' type='boolean' value='false'/>
+    <property name='nrip_save' type='boolean' value='false'/>
+    <property name='sse4a' type='boolean' value='true'/>
+    <property name='ssse3' type='boolean' value='true'/>
+    <property name='kvm_pv_unhalt' type='boolean' value='false'/>
+    <property name='invpcid' type='boolean' value='false'/>
+    <property name='pdpe1gb' type='boolean' value='true'/>
+    <property name='tsc-deadline' type='boolean' value='false'/>
+    <property name='fma' type='boolean' value='false'/>
+    <property name='cx16' type='boolean' value='true'/>
+    <property name='de' type='boolean' value='true'/>
+    <property name='enforce' type='boolean' value='false'/>
+    <property name='stepping' type='number' value='3'/>
+    <property name='xsave' type='boolean' value='true'/>
+    <property name='clflush' type='boolean' value='true'/>
+    <property name='skinit' type='boolean' value='false'/>
+    <property name='tce' type='boolean' value='false'/>
+    <property name='tsc' type='boolean' value='true'/>
+    <property name='fpu' type='boolean' value='true'/>
+    <property name='ds-cpl' type='boolean' value='false'/>
+    <property name='ds_cpl' type='boolean' value='false'/>
+    <property name='ibs' type='boolean' value='false'/>
+    <property name='host-phys-bits' type='boolean' value='false'/>
+    <property name='fma4' type='boolean' value='false'/>
+    <property name='la57' type='boolean' value='true'/>
+    <property name='osvw' type='boolean' value='false'/>
+    <property name='check' type='boolean' value='true'/>
+    <property name='hv-spinlocks' type='number' value='-1'/>
+    <property name='pmm' type='boolean' value='false'/>
+    <property name='apic' type='boolean' value='true'/>
+    <property name='pmu' type='boolean' value='false'/>
+    <property name='min-xlevel2' type='number' value='0'/>
+    <property name='tsc-adjust' type='boolean' value='false'/>
+    <property name='tsc_adjust' type='boolean' value='false'/>
+    <property name='kvm-steal-time' type='boolean' value='false'/>
+    <property name='kvm_steal_time' type='boolean' value='false'/>
+    <property name='kvmclock' type='boolean' value='false'/>
+    <property name='l3-cache' type='boolean' value='true'/>
+    <property name='lwp' type='boolean' value='false'/>
+    <property name='xop' type='boolean' value='false'/>
+    <property name='avx' type='boolean' value='false'/>
+    <property name='ospke' type='boolean' value='true'/>
+    <property name='ace2' type='boolean' value='false'/>
+    <property name='acpi' type='boolean' value='true'/>
+    <property name='avx512bw' type='boolean' value='false'/>
+    <property name='hv-vapic' type='boolean' value='false'/>
+    <property name='fsgsbase' type='boolean' value='true'/>
+    <property name='ht' type='boolean' value='false'/>
+    <property name='nx' type='boolean' value='true'/>
+    <property name='pclmulqdq' type='boolean' value='true'/>
+    <property name='mmxext' type='boolean' value='true'/>
+    <property name='popcnt' type='boolean' value='true'/>
+    <property name='xsaves' type='boolean' value='false'/>
+    <property name='lm' type='boolean' value='true'/>
+    <property name='umip' type='boolean' value='false'/>
+    <property name='avx2' type='boolean' value='false'/>
+    <property name='pse' type='boolean' value='true'/>
+    <property name='sep' type='boolean' value='true'/>
+    <property name='pclmuldq' type='boolean' value='true'/>
+    <property name='nodeid-msr' type='boolean' value='false'/>
+    <property name='kvm' type='boolean' value='true'/>
+    <property name='misalignsse' type='boolean' value='false'/>
+    <property name='min-xlevel' type='number' value='2147483658'/>
+    <property name='bmi2' type='boolean' value='true'/>
+    <property name='bmi1' type='boolean' value='true'/>
+    <property name='kvm-pv-unhalt' type='boolean' value='false'/>
+    <property name='realized' type='boolean' value='false'/>
+    <property name='tsc_scale' type='boolean' value='false'/>
+    <property name='tsc-scale' type='boolean' value='false'/>
+    <property name='topoext' type='boolean' value='false'/>
+    <property name='hv-vpindex' type='boolean' value='false'/>
+    <property name='xlevel2' type='number' value='0'/>
+    <property name='clflushopt' type='boolean' value='true'/>
+    <property name='monitor' type='boolean' value='true'/>
+    <property name='avx512er' type='boolean' value='false'/>
+    <property name='pmm-en' type='boolean' value='false'/>
+    <property name='pcid' type='boolean' value='false'/>
+    <property name='3dnow' type='boolean' value='true'/>
+    <property name='erms' type='boolean' value='true'/>
+    <property name='lahf-lm' type='boolean' value='true'/>
+    <property name='lahf_lm' type='boolean' value='true'/>
+    <property name='xstore' type='boolean' value='false'/>
+    <property name='hv-synic' type='boolean' value='false'/>
+    <property name='fxsr-opt' type='boolean' value='false'/>
+    <property name='fxsr_opt' type='boolean' value='false'/>
+    <property name='rtm' type='boolean' value='false'/>
+    <property name='lmce' type='boolean' value='false'/>
+    <property name='hv-time' type='boolean' value='false'/>
+    <property name='perfctr-nb' type='boolean' value='false'/>
+    <property name='perfctr_nb' type='boolean' value='false'/>
+    <property name='ffxsr' type='boolean' value='false'/>
+    <property name='rdrand' type='boolean' value='false'/>
+    <property name='rdseed' type='boolean' value='false'/>
+    <property name='avx512-4vnniw' type='boolean' value='false'/>
+    <property name='vme' type='boolean' value='false'/>
+    <property name='vmx' type='boolean' value='false'/>
+    <property name='dtes64' type='boolean' value='false'/>
+    <property name='mtrr' type='boolean' value='true'/>
+    <property name='rdtscp' type='boolean' value='true'/>
+    <property name='pse36' type='boolean' value='true'/>
+    <property name='tbm' type='boolean' value='false'/>
+    <property name='wdt' type='boolean' value='false'/>
+    <property name='pause_filter' type='boolean' value='false'/>
+    <property name='model-id' type='string' value='QEMU TCG CPU version 2.5+'/>
+    <property name='sha-ni' type='boolean' value='false'/>
+    <property name='abm' type='boolean' value='true'/>
+    <property name='avx512pf' type='boolean' value='false'/>
+    <property name='xstore-en' type='boolean' value='false'/>
+  </hostCPU>
+  <cpu type='kvm' name='max' usable='yes'/>
+  <cpu type='kvm' name='host' usable='yes'/>
+  <cpu type='kvm' name='base' usable='yes'/>
+  <cpu type='kvm' name='qemu64' usable='yes'/>
+  <cpu type='kvm' name='qemu32' usable='yes'/>
+  <cpu type='kvm' name='phenom' usable='no'/>
+  <cpu type='kvm' name='pentium3' usable='yes'/>
+  <cpu type='kvm' name='pentium2' usable='yes'/>
+  <cpu type='kvm' name='pentium' usable='yes'/>
+  <cpu type='kvm' name='n270' usable='yes'/>
+  <cpu type='kvm' name='kvm64' usable='yes'/>
+  <cpu type='kvm' name='kvm32' usable='yes'/>
+  <cpu type='kvm' name='coreduo' usable='yes'/>
+  <cpu type='kvm' name='core2duo' usable='yes'/>
+  <cpu type='kvm' name='athlon' usable='no'/>
+  <cpu type='kvm' name='Westmere' usable='yes'/>
+  <cpu type='kvm' name='Skylake-Client' usable='yes'/>
+  <cpu type='kvm' name='SandyBridge' usable='yes'/>
+  <cpu type='kvm' name='Penryn' usable='yes'/>
+  <cpu type='kvm' name='Opteron_G5' usable='no'/>
+  <cpu type='kvm' name='Opteron_G4' usable='no'/>
+  <cpu type='kvm' name='Opteron_G3' usable='no'/>
+  <cpu type='kvm' name='Opteron_G2' usable='yes'/>
+  <cpu type='kvm' name='Opteron_G1' usable='yes'/>
+  <cpu type='kvm' name='Nehalem' usable='yes'/>
+  <cpu type='kvm' name='IvyBridge' usable='yes'/>
+  <cpu type='kvm' name='Haswell' usable='yes'/>
+  <cpu type='kvm' name='Haswell-noTSX' usable='yes'/>
+  <cpu type='kvm' name='Conroe' usable='yes'/>
+  <cpu type='kvm' name='Broadwell' usable='yes'/>
+  <cpu type='kvm' name='Broadwell-noTSX' usable='yes'/>
+  <cpu type='kvm' name='486' usable='yes'/>
+  <cpu type='tcg' name='max' usable='yes'/>
+  <cpu type='tcg' name='host' usable='no'/>
+  <cpu type='tcg' name='base' usable='yes'/>
+  <cpu type='tcg' name='qemu64' usable='yes'/>
+  <cpu type='tcg' name='qemu32' usable='yes'/>
+  <cpu type='tcg' name='phenom' usable='no'/>
+  <cpu type='tcg' name='pentium3' usable='yes'/>
+  <cpu type='tcg' name='pentium2' usable='yes'/>
+  <cpu type='tcg' name='pentium' usable='yes'/>
+  <cpu type='tcg' name='n270' usable='yes'/>
+  <cpu type='tcg' name='kvm64' usable='yes'/>
+  <cpu type='tcg' name='kvm32' usable='yes'/>
+  <cpu type='tcg' name='coreduo' usable='yes'/>
+  <cpu type='tcg' name='core2duo' usable='yes'/>
+  <cpu type='tcg' name='athlon' usable='yes'/>
+  <cpu type='tcg' name='Westmere' usable='yes'/>
+  <cpu type='tcg' name='Skylake-Client' usable='no'/>
+  <cpu type='tcg' name='SandyBridge' usable='no'/>
+  <cpu type='tcg' name='Penryn' usable='yes'/>
+  <cpu type='tcg' name='Opteron_G5' usable='no'/>
+  <cpu type='tcg' name='Opteron_G4' usable='no'/>
+  <cpu type='tcg' name='Opteron_G3' usable='no'/>
+  <cpu type='tcg' name='Opteron_G2' usable='yes'/>
+  <cpu type='tcg' name='Opteron_G1' usable='yes'/>
+  <cpu type='tcg' name='Nehalem' usable='yes'/>
+  <cpu type='tcg' name='IvyBridge' usable='no'/>
+  <cpu type='tcg' name='Haswell' usable='no'/>
+  <cpu type='tcg' name='Haswell-noTSX' usable='no'/>
+  <cpu type='tcg' name='Conroe' usable='yes'/>
+  <cpu type='tcg' name='Broadwell' usable='no'/>
+  <cpu type='tcg' name='Broadwell-noTSX' usable='no'/>
+  <cpu type='tcg' name='486' usable='yes'/>
+  <machine name='pc-i440fx-2.9' alias='pc' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='pc-0.12' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='pc-i440fx-2.4' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='pc-1.3' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='pc-q35-2.7' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='pc-q35-2.6' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='xenpv' maxCpus='1'/>
+  <machine name='pc-i440fx-1.7' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='pc-i440fx-2.8' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='pc-i440fx-1.6' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='pc-i440fx-2.7' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='pc-0.11' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='pc-i440fx-2.3' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='pc-0.10' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='pc-1.2' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='pc-i440fx-2.2' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='isapc' hotplugCpus='yes' maxCpus='1'/>
+  <machine name='pc-q35-2.5' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='xenfv' hotplugCpus='yes' maxCpus='128'/>
+  <machine name='pc-0.15' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='pc-0.14' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='pc-i440fx-1.5' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='pc-i440fx-2.6' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='pc-i440fx-1.4' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='pc-i440fx-2.5' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='pc-q35-2.9' alias='q35' hotplugCpus='yes' maxCpus='288'/>
+  <machine name='pc-1.1' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='pc-i440fx-2.1' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='pc-q35-2.8' hotplugCpus='yes' maxCpus='288'/>
+  <machine name='pc-1.0' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='pc-i440fx-2.0' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='pc-q35-2.4' hotplugCpus='yes' maxCpus='255'/>
+  <machine name='pc-0.13' hotplugCpus='yes' maxCpus='255'/>
+</qemuCaps>
